lib/fast_jsonapi/object_serializer.rb

```diff
@@ -41,8 +42,8 @@ def hash_for_one_record
 
       return serializable_hash unless @resource
 
-      serializable_hash[:data] = self.class.record_hash(@resource, @params)
-      serializable_hash[:included] = self.class.get_included_records(@resource, @includes, @known_included_objects, @params) if @includes.present?
+      serializable_hash[:data] = self.class.record_hash(@resource, @fieldsets[self.class.reflected_record_type.to_sym], @params)
```

Comment:
`reflected_record_type` is the meaningful default it can be overridden in `set_type` so it probably makes sense here to use `self.record_type`

Reply:
👍